core/src/OpenACC/Kokkos_OpenACC_Exec.hpp

```diff
@@ -0,0 +1,5300 @@
+    }
+
+  inline
+  void team_fan_out() const
+    {
+      memory_fence();
+      for ( int n = 1 , j ; ( ( j = m_team_rank_rev + n ) < m_team_size ) && ! (
+  m_team_rank_rev & n ) ; n <<= 1 ) { m_exec.pool_rev( m_team_base_rev + j
+  )->state_set( Active ); memory_fence();
+      }
+    }
+  */
+ public:
+  KOKKOS_INLINE_FUNCTION
+  const execution_space::scratch_memory_space& team_shmem() const {
+    return m_team_shared.set_team_thread_mode(0, 1, 0);
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  const execution_space::scratch_memory_space& team_scratch(int level) const {
+    return m_team_shared.set_team_thread_mode(level, 1,
+                                              m_team_scratch_size[level]);
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  const execution_space::scratch_memory_space& thread_scratch(int level) const {
+    return m_team_shared.set_team_thread_mode(level, team_size(), team_rank());
+  }
+
+  KOKKOS_INLINE_FUNCTION int league_rank() const { return m_league_rank; }
+  KOKKOS_INLINE_FUNCTION int league_size() const { return m_league_size; }
+  KOKKOS_INLINE_FUNCTION int team_rank() const { 
+#ifdef __USE_NVHPC__
+	if (m_team_size <= 1) {
+		return m_team_rank;
+	} else {
+    	return __pgi_vectoridx(); 
+	}
+#else
+    return m_team_rank; 
+#endif
+  }
+  KOKKOS_INLINE_FUNCTION int vector_length() const { return m_vector_length; }
+  KOKKOS_INLINE_FUNCTION int team_size() const { return m_team_size; }
+  KOKKOS_INLINE_FUNCTION void* impl_reduce_scratch() const {
+    return m_reduce_scratch;
+  }
+
+  KOKKOS_INLINE_FUNCTION void team_barrier() const {
+//#pragma omp barrier
+	//[FIXME-SL] OpenACC does not provide any explicit barrier constructs for device kernels.
+	std::cerr << "Kokkos::Experimental::OpenACC ERROR: OpenACC does not provide any explicit barrier constructs for device kernels; exit!" << std::endl;
+	exit(0);
+  }
+
+  template <class ValueType>
+  KOKKOS_INLINE_FUNCTION void team_broadcast(ValueType& value,
+                                             int thread_id) const {
+    // Make sure there is enough scratch space:
+    using type =
+        typename std::conditional<(sizeof(ValueType) < TEAM_REDUCE_SIZE),
+                                  ValueType, void>::type;
+    type* team_scratch = reinterpret_cast<type*>(
+        ((char*)(m_glb_scratch) + TEAM_REDUCE_SIZE * league_rank()));
+//#pragma omp barrier
+    if (team_rank() == thread_id) *team_scratch = value;
+//#pragma omp barrier
+    value = *team_scratch;
+  }
+
+  template <class Closure, class ValueType>
+  KOKKOS_INLINE_FUNCTION void team_broadcast(const Closure& f, ValueType& value,
+                                             const int& thread_id) const {
+    f(value);
+    team_broadcast(value, thread_id);
+  }
+
+  template <class ValueType, class JoinOp>
+  KOKKOS_INLINE_FUNCTION ValueType team_reduce(const ValueType& value,
+                                               const JoinOp& op_in) const {
+//#pragma omp barrier
```

Comment:
Why do you have `#pragma omp` here and below?

Reply:
We had kept `#pragma omp barrier` as a placeholder to be replaced with a barrier statement.
The current OpenACC standard does not support explicit barrier instructions in a compute region, and thus correctly implementing these codes will be a future work.
For clarification, `//#pragma amp barrier` is replaced with the following:
`//FIXME_OPENACC`
 `//#pragma acc barrier`